pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1170,25 +1185,56 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 
 	// convert to transport socket matcher if the mode was auto detected
 	if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL && mtlsCtxType == autoDetected && util.IsIstioVersionGE14(proxy) {
-		tlsContext, err := ptypes.MarshalAny(cluster.TlsContext)
-		cluster.TlsContext = nil
+		istioTLSContext, err := ptypes.MarshalAny(cluster.TlsContext)
+		if err != nil {
+			log.Errorf("error marshaling istio tls context to transport_socket config for cluster %s, err=%v",
+				cluster.Name, err)
+			return // no tls context for the cluster
+		}
+
+		simpleTLSContext, err := ptypes.MarshalAny(&auth.UpstreamTlsContext{
+			CommonTlsContext: &auth.CommonTlsContext{
+				ValidationContextType: &auth.CommonTlsContext_ValidationContext{
+					// This is empty. If the user wanted CA verification, then a destination rule would have been
```

Comment:
We do need trusted_ca to validate server cert, right?

Reply:
No. that beats the purpose of automatic fallback. people can then define dest rules if they want